lib/rules/no-obscure-array-access.js

```diff
@@ -0,0 +1,79 @@
+import { builders } from 'ember-template-recast';
+
+import Rule from './_base.js';
+
+export function ERROR_MESSAGE() {
+  return `Obscure expressions are prohibited. Please use Ember's get helper instead. e.g. {{get @list '0'}} `;
+}
+
+function buildParams(node) {
```

Comment:
Can you add commenting all throughout this file explaining what is going on at each step? The logic is a bit complex.

Reply:
Thanks for adding the comments. It's extremely valuable to have the autofixer. The logic can be further simplified later but it's important to have comments and solid tests now.

Reply:
Cool, I've simplified the fixer now so it should be a littler more straightforward.

Reply:
Wow, nice!